src/extensions/ThumbnailsExtension/ThumbnailsExtensionBifParser.js

```diff
@@ -0,0 +1,171 @@
+    this.numberOfBIFImages = this.data.getUint32(NUMBER_OF_BIF_IMAGES_OFFSET, true);
+
+    // Version
+    // SEE: https://sdkdocs.roku.com/display/sdkdoc/Trick+Mode+Support#TrickModeSupport-Version
+    this.version = this.data.getUint32(VERSION_OFFSET, true);
+
+    this.bifIndex = this.generateBIFIndex(true);
+  }
+
+  /**
+   * Create the BIF index
+   * SEE: https://sdkdocs.roku.com/display/sdkdoc/Trick+Mode+Support#TrickModeSupport-BIFindex
+   *
+   * @returns {Array} bifIndex
+   */
+  generateBIFIndex() {
+    const bifIndex = [];
+
+    for (
+      // BIF index starts at byte 64 (BIF_INDEX_OFFSET)
+      let i = 0, bifIndexEntryOffset = BIF_INDEX_OFFSET;
+      i < this.numberOfBIFImages;
+      i += 1, bifIndexEntryOffset += BIF_INDEX_ENTRY_LENGTH
+    ) {
+      const bifIndexEntryTimestampOffset = bifIndexEntryOffset;
+      const bifIndexEntryAbsoluteOffset = bifIndexEntryOffset + 4;
+
+      const nextBifIndexEntryAbsoluteOffset = bifIndexEntryAbsoluteOffset + BIF_INDEX_ENTRY_LENGTH;
+
+      // Documented example, items within `[]`are used to generate the frame.
+      // 64, 65, 66, 67 | 68, 69, 70, 71
+      // [Frame 0 timestamp] | [absolute offset of frame]
+      // 72, 73, 74, 75 | 76, 77, 78, 79
+      // Frame 1 timestamp | [absolute offset of frame]
+      const offset = this.data.getUint32(bifIndexEntryAbsoluteOffset, true);
+      const nextOffset = this.data.getUint32(nextBifIndexEntryAbsoluteOffset, true);
+      const timestamp = this.data.getUint32(bifIndexEntryTimestampOffset, true);
+
+      bifIndex.push({
+        offset,
+        timestamp,
+
+        length: nextOffset - offset,
+      });
+    }
+
+    return bifIndex;
+  }
+
+  /**
+   * Return image data for a specific frame of a movie.
+   *
+   * @param {number} second
+   * @returns {string} imageData
+   */
+  getImageDataAtSecond(second) {
+    const image = 'data:image/jpeg;base64,';
+
+    // since frames are defined at an interval of `this.framewiseSeparation`,
+    // we need to convert the time into an appropriate frame number.
+    const frameNumber = Math.floor(second / (this.framewiseSeparation / 1000));
+
+    const frame = this.bifIndex[frameNumber];
+
+    console.log({frame});
+
+    if (!frame) {
+      return image;
+    }
+
+    return `${image}${btoa(String.fromCharCode.apply(null,
+      new Uint8Array(this.arrayBuffer.slice(frame.offset, frame.offset + frame.length))
+    ))}`;
+  }
+
+  /**
+   * Return image data for all BIF files.
+   *
+   * @returns {string} imageData
+   */
+  getImageData() {
+    const images = [];
+    this.bifIndex.forEach(frame => {
+      const image = 'data:image/jpeg;base64,';
+
+      images.push({
+        start: frame.timestamp / 1000,
+        src: `${image}${btoa(String.fromCharCode.apply(null,
+            new Uint8Array(this.arrayBuffer.slice(frame.offset, frame.offset + frame.length))
+          ))}`,
+        x: 0,
+        y: 0,
+        width: 240,
```

Comment:
width & height are fixed, will this break for different BIF files or where does these values come from?

Reply:
I also noticed that, and decided that while it probably isn't the biggest problem, we should figure that out.  So I set up a [default value](https://github.com/matvp91/indigo-player/compare/master...stoicbuddha:bifupdates?diff=unified&expand=1#diff-a7e5392951c4211ea2d653559292f8d9R76) for dimensions and then try to [get dimensions](https://github.com/matvp91/indigo-player/compare/master...stoicbuddha:bifupdates?diff=unified&expand=1#diff-a7e5392951c4211ea2d653559292f8d9R79) for the first BIF file in the array. If we get dimensions, we overwrite the defaults, and if we have an issue, we run with the defaults.